endoreg_db/serializers/label_video_segment/_lvs_update.py

```
@@ -0,0 +1,138 @@
+
+    # Final validation of frame numbers
+    if instance.start_frame_number >= instance.end_frame_number:
+        raise serializers.ValidationError("start_frame must be less than end_frame.")
+    
+    if instance.video and instance.end_frame_number > instance.video.frame_count:
+        raise serializers.ValidationError("end_frame cannot exceed the total frames of the video.")
+
+    instance.save()
+    return instance
+
+def _update(self:"LabelVideoSegmentSerializer", instance, validated_data):
```

Pylint (3.3.7)

[refactor] Too many branches (13/12)

(R0912)